catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -70,6 +70,18 @@
 
     private static final String DISTANCE_SORT_FIELD = "_distance_";
 
+    private static final String SOLR_QUERY_ROWS_PROPERTY = "solr.query.rows";
+
+    /**
+     * Because {@link SolrQuery#setRows(Integer)} will increment the given integer and use it as an
+     * array index in Solr Cloud mode, we cannot use {@link Integer#MAX_VALUE} by default and should
+     * pick a much lower, sensible ceiling for our row count. In practice, an un-paged query with a
```

Comment:
`MAX_VALUE` minus some constant may crash the front end for significantly large catalog deployments in cases where our API is used without paging. Remember this number represents the max size of a single page; but even then, query results this large arguably are not useful. No one is going to scan that many records. 

500k was somewhat arbitrary based on reasonable page sizes that browsers can handle at once. Also, assuming the average metacard is 3 KB, which is an under-estimate to begin with, that would make a paging response with 500,000 results to be about 1.5 GB which is already way past what most browsers have in terms of default memory allocation.